### PR TITLE
Clean the debounce of getLayerHistogram

### PR DESCRIPTION
**Pull request by bruyeret**
Reformat getLayerHistogram to make it clearer.
Fix an infinite recursion.

Closes [#310](https://github.com/Kitware/UPennContrast/issues/310)

### CODE REVIEW

File: src/store/index.ts [hunk 1]
```diff
@@ -932,34 +932,50 @@ export class Main extends VuexModule {
...
+            histogramObj.lastHistogram = value;
+          });
+          histogramObj.promise.catch(() => {
+            histogramObj.lastHistogram = null;
+          });
+          histogramObj.promise.finally(() => {
+            histogramObj.lastImages = images;
+            histogramObj.nextImages = null;
+            histogramObj.lock = false;
+            nextHistogram();
```
Comment (luciemac) [inline src/store/index.ts:964]: You can write
```
 histogramObj.promise
    .then(...)
    .catch(...);
    .finally(...);
```

File: src/store/index.ts [hunk 2]
```diff
@@ -932,34 +932,50 @@ export class Main extends VuexModule {
...
-        layer._histogram.next = images;
-        if (layer._histogram.last) {
-          nextHistogram();
-        }
+
+      const imagesToId = (images: IImage[] | null) => {
+        return (
+          images?.map(i => `${i.item._id}#${i.frameIndex}`).join(",") || "null"
+        );
+      };
```
Comment (luciemac) [inline src/store/index.ts:974]: `const imagesToId = (images: IImage[] | null) => images?.map(i =>`${i.item._id}#${i.frameIndex}`).join(",") || null;`

File: src/store/index.ts [hunk 3]
```diff
@@ -932,34 +932,50 @@ export class Main extends VuexModule {
...
       };
-      if (images !== layer._histogram.images) {
-        layer._histogram.next = images;
-        if (layer._histogram.last) {
-          nextHistogram();
-        }
+
+      const imagesToId = (images: IImage[] | null) => {
+        return (
+          images?.map(i => `${i.item._id}#${i.frameIndex}`).join(",") || "null"
```
Comment (luciemac) [inline src/store/index.ts:972]: why do you return a string "null" and not just "null"?

Comment (bruyeret) [reply src/store/index.ts:972]: This is made so that `imagesToId` always return a string.

Comment (luciemac) [reply src/store/index.ts]: what about an empty string?

Comment (bruyeret) [reply src/store/index.ts]: An empty string could be returned with `null` or `[]` as arguments

**Comment by bruyeret** [comment]
This PR fixes issue #310 but the bug was already fixed by PR #309 which fixed issue #307.
This PR should be merged anyway because it refactors and clarifies this part of the code.
I kept the logic of PR #309.

**Comment by luciemac** [comment]
LGTM

**Pull request merged**